### DISCUSSION TITLE
Help: OpenId with SPA

### DISCUSSION THREAD

**Discussion by JyothishGeorge** [OrchardCMS/OrchardCore#9175] [Q&A]
Hi  

I am trying to setup OpenId connect between an angular single page application and Orchard core (api) server. I'm a totally new to the identityserver and openid, gone through the documentation and related videos to understand it in detail. The client application sends a request to connect and authorise as below,  

https://localhost:44342/connect/authorize?client_id=code_flow_client_id&redirect_uri=https%3A%2F%2Flocalhost%3A4200%2Fsignin-callback&response_type=code&scope=openid%20profile%20api&state=ecd0dbd219ce4bc581a05db8fd4bcd30&code_challenge=ss4i5MOZOLJ_Js1-YsTYPHjf3YR1l5boqq3yTiaiJIE&code_challenge_method=S256&response_mode=query 

The identityserver returns back an error "unauthorized_client The specified 'response_type' is not valid for this client application". The server is expected to redirect to the login page instead it shows general exception page. The log says "The token request was rejected because the client application was not found: 'code_flow_client_id'". My open Id settings are follows, the application I set up in the server has the client id provided, now I am unable to locate the issue, any help on this is highly appreciated.  

### **Configure OpenID server settings** 

**Endpoints** 

✔ Enable Token Endpoint 

✔ Enable Authorization Endpoint 

✔ Enable Logout Endpoint 

✔ Enable User Info Endpoint 

**Flows** 

✔ Allow Authorization Code Flow 

Allow Implicit Flow 

Allow Hybrid Flow 

✔ Allow Password Flow 

✔ Allow Refresh Token Flow 

✔ Allow Client Credentials Flow 

Authority: https://localhost:44342/ 

Encryption Certificate Store Location:    None 

Signing Certificate Store Location:   None      

Token Format: JSON Web Token (JWT) 

### Configure OpenID validation settings 

 

Authorization server tenant None 

Authority: https://localhost:44342/ 

Audience: my_recipient 

 

### **Edit an application** 

Client Id: code_flow_client_id 

Display Name: Authorization Code Flow 

Type: Public Client 

 
**Flows** 

✔ Allow Authorization Code Flow 

Redirect Uris: https://localhost:4200/signin-callback 

✔ Allow Logout Endpoint: https://localhost:4200/signout-callback 

Consent type: Explicit Consent 

### **Edit a scope** 

Name:api 

Display Name: api Scope 

Tenants 

Additional resources (API audiences): my_recipient

**Comment by EricBui619** [comment]
I guess problem comes from your client setup. I provide you my configurations for Open Id. After setup, I use postman to make a login request. So first u can check and compare my configurations with the one of you for detecting what info your setup is lacking of or wrong

Client setup:
![image](https://user-images.githubusercontent.com/30250345/114852206-c2cdde80-9e0c-11eb-8635-3b46aff0f041.png)

OpenId server setup:
![image](https://user-images.githubusercontent.com/30250345/114852360-e3963400-9e0c-11eb-9807-f05089a5c102.png)

Postman parameters:
![image](https://user-images.githubusercontent.com/30250345/114852439-fa3c8b00-9e0c-11eb-97bf-423d81aef832.png)

**Comment by EricBui619** [comment]
Please note that: your client is public so the secret key(code_challenge) is not necessary

**Reply by JyothishGeorge** [reply]
Hi EricArita,

Thanks a lot for sharing the screenshots, it was really helpfull, its up and running now, for some reason the client id was not correctly  set, I tried with postman as in the screenshot and its working. I litrerally transferred the configuration back to client,

Thanks a lot,
